### DISCUSSION TITLE
https://raw.githubusercontent.com sends back a 404 HTTP code

### DISCUSSION THREAD

**Discussion by t0pd4wn** [community/community#53538] [Repositories]
### Select Topic Area

Bug

### Body

Hello,

I've been witnessing a weird behavior with raw.githubusercontent, where some URLs sends back a 404 code, although content is available. I had the 404 with both wget and browser. Navigating back through the GUI from the repository root to the file seemed to "refresh" the raw.githubusercontent link, but wget is still inconsistent.

Here is the link in question :
https://raw.githubusercontent.com/t0pd4wn/loco.sh/gh-main/profiles/full/dotfiles/.p10k.zsh

Kind regards,

**Comment by koppor** [comment]
Maybe, GitHub has some spam protection in place.

I think, these days, the behavior changed again. I have troubles with `jbang`, which relies on `jbang-catalog.json` of a repository.

**Comment by ericthelizard** [comment]
Having a similar problem with https://raw.githubusercontent.com/VictoriaMetrics/VictoriaMetrics/refs/tags/v1.114.0/LICENSE

**Comment by micafer** [comment]
I have this error intermittently:
https://raw.githubusercontent.com/grycap/tosca/refs/heads/main/templates/simple-node-disk.yml

**Comment by mvink-guida** [comment]
similar issues
https://github.com/cilium/cilium/blob/1.16.3/pkg/k8s/apis/cilium.io/client/crds/v2/ciliumnetworkpolicies.yaml
--> 
https://raw.githubusercontent.com/cilium/cilium/refs/tags/1.16.3/pkg/k8s/apis/cilium.io/client/crds/v2/ciliumnetworkpolicies.yaml 404
on main it works

another example on main doesnt work

https://github.com/kubernetes-sigs/gateway-api/blob/main/config/crd/standard/gateway.networking.k8s.io_gatewayclasses.yaml
-->
https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/refs/heads/main/config/crd/standard/gateway.networking.k8s.io_gatewayclasses.yaml 404

**Comment by timofurrer** [comment]
Same for https://github.com/bufbuild/protovalidate/blob/main/proto/protovalidate/buf/validate/validate.proto - failing all our builds because of this.

**Comment by liamjones** [comment]
Started getting this today on https://raw.githubusercontent.com/react-native-community/rn-diff-purge/refs/heads/master/RELEASES which broke https://react-native-community.github.io/upgrade-helper/?from=0.74.6&to=0.79.5 (which I've been using fine this last week)

Seems to apply to other files in the repo too: https://raw.githubusercontent.com/react-native-community/rn-diff-purge/release/0.79.5/RnDiffApp/android/gradlew.bat

**Comment by SpangleLabs** [comment]
Another example here, this file exists in the repo:
https://github.com/grpc-ecosystem/grpc-gateway/blob/v1.14.6/third_party/googleapis/google/api/annotations.proto
But clicking raw download gives a 404 error:
https://raw.githubusercontent.com/grpc-ecosystem/grpc-gateway/refs/tags/v1.14.6/third_party/googleapis/google/api/annotations.proto

Seems like it's working in some repositories, and not in others. But seems repository-wide when it's happening

**Comment by AlexanderMann** [comment]
Same issue here at Apollo GraphQL.

*Tagged github hosted file:*
https://github.com/apollographql/rover/blob/v0.35.0/installers/binstall/scripts/nix/install.sh

*404 when clicking through to the raw file:*
https://raw.githubusercontent.com/apollographql/rover/refs/tags/v0.35.0/installers/binstall/scripts/nix/install.sh

*A tagged raw file which is being served...but _pointing to_ the wrong file:*
https://raw.githubusercontent.com/apollographql/rover/v0.35.0/installers/binstall/scripts/nix/install.sh

⬆️  Note that in there, instead of that being `0.35` it's `0.14`.

We're actually seeing that same behaviour with a lot of our deploys.

📆 If timestamps are useful, this started being reported at Apollo around `2025-08-08 04:27 UTC`

**Reply by AlexanderMann** [reply]
Apollo's internal alerts are showing _our_ problems are resolved as of ~10 minutes ago. We're unsure if this will hold, but wanted to help update the community 😅

**Reply by AlexanderMann** [reply]
Got a response from GH noting that the errant code has been rolled back for the 404s.

**Comment by AlexanderMann** [comment]
For those who end up here: https://github.com/aquaproj/aqua/issues/4068#issuecomment-3166806915
